lib/rules/prefer-arrow-callback.js

```diff
@@ -308,6 +310,12 @@ module.exports = {
 								return;
 							}
 
+							if (
+								node.params.some(param => param.name === "this")
+							) {
+								return;
+							}
```

Comment:
[Refactor] TypeScript only allows the first param to be named `this`. JavaScript doesn't allow any params to be named `this`. This can just check the first param, if it exists.